### PR TITLE
Export DefsFactory at dagster_airlift.def_factory

### PR DESCRIPTION
**Pull request by schrockn**
## Summary & Motivation

Export DefsFactory at dagster_airlift.core

## How I Tested These Changes

BK

### CODE REVIEW

**Comment by schrockn** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/23564?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#23564** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/23564?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#23533** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/23533?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @schrockn and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by dpeng817 — APPROVED** [review]

File: examples/experimental/dagster-airlift/README.md [hunk 1]
```diff
@@ -119,7 +119,7 @@ For most projects we anticipate the following process:
 In our example we have a pre-written factory class (in `examples/experimental/dagster-airlift/dagster_airlift/dbt/multi_asset.py` and installable by `uv pip install dagster-airlift[dbt]`) that is associated with invoking a dbt project:
 
 ```python
-from dagster_airlift.core.def_factory import DefsFactory
```
Comment (dpeng817) [inline examples/experimental/dagster-airlift/README.md:122]: should be airlift.core not airlift.def_factory right?

**Review by dpeng817 — APPROVED** [review]
see comment

**Comment by schrockn** [comment]
### Merge activity

* **Aug 9, 6:22 PM EDT**: @schrockn started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/23564).

**Pull request merged**